homeassistant/components/withings/common.py

```diff
@@ -283,61 +797,308 @@ async def update_sleep_summary(self) -> SleepGetSummaryResponse:
+            hass,
+            profile,
+            ConfigEntryWithingsApi(
+                hass=hass,
+                config_entry=config_entry,
+                implementation=await config_entry_oauth2_flow.async_get_config_entry_implementation(
+                    hass, config_entry
+                ),
+            ),
+            config_entry.data["token"]["userid"],
+            WebhookConfig(
+                id=config_entry.data[CONF_WEBHOOK_ID],
+                url=config_entry.data[const.CONF_WEBHOOK_URL],
+                enabled=config_entry.data[const.CONF_USE_WEBHOOK],
+            ),
+        )
 
-def create_withings_data_manager(
-    hass: HomeAssistant,
-    config_entry: ConfigEntry,
-    implementation: AbstractOAuth2Implementation,
-) -> WithingsDataManager:
-    """Set up the sensor config entry."""
-    profile = config_entry.data.get(const.PROFILE)
-
-    _LOGGER.debug("Creating withings api instance")
-    api = ConfigEntryWithingsApi(
-        hass=hass, config_entry=config_entry, implementation=implementation
+    return config_entry_data[const.DATA_MANAGER]
+
+
+def get_data_manager_by_user_id(
+    hass: HomeAssistant, user_id: int
+) -> Optional[DataManager]:
+    """Get a data manager by the user id."""
+    return next(
+        iter(
+            [
+                data_manager
+                for data_manager in get_all_data_managers(hass)
+                if data_manager.user_id == user_id
+            ]
+        ),
+        None,
     )
 
-    _LOGGER.debug("Creating withings data manager for profile: %s", profile)
-    return WithingsDataManager(hass, profile, api)
+
+def get_data_manager_by_webhook_id(
+    hass: HomeAssistant, webhook_id: str
+) -> Optional[DataManager]:
+    """Get a data manager by it's webhook id."""
+    return next(
+        iter(
+            [
+                data_manager
+                for data_manager in get_all_data_managers(hass)
+                if data_manager.webhook_config.id == webhook_id
+            ]
+        ),
+        None,
+    )
 
 
-def get_data_manager(
-    hass: HomeAssistant,
-    entry: ConfigEntry,
-    implementation: AbstractOAuth2Implementation,
-) -> WithingsDataManager:
-    """Get a data manager for a config entry.
+def get_all_data_managers(hass: HomeAssistant) -> Tuple[DataManager, ...]:
+    """Get all configured data managers."""
+    return tuple(
+        [
+            config_entry_data[const.DATA_MANAGER]
+            for config_entry_data in hass.data[const.DOMAIN].values()
+            if const.DATA_MANAGER in config_entry_data
+        ]
+    )
 
-    If the data manager doesn't exist yet, it will be
-    created and cached for later use.
-    """
-    entry_id = entry.entry_id
 
-    hass.data[const.DOMAIN] = hass.data.get(const.DOMAIN, {})
+def async_remove_data_manager(hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+    """Stop and remove a data manager for a config entry."""
```

Comment:
Correct, this is intentional. The hass.data dictionaries are not strongly typed. So keeping the set/get code close to each other makes it easier to manage when it comes time to change it. This also prevents once from accidentally using hass.data instead of calling the get method.

Reply:
Ok then please update the docstring and remove the stopping part.